src/components/Todo/EditCard/edit-card.jsx

```diff
@@ -0,0 +1,65 @@
+import React, { useState, useEffect, useRef } from "react";
+import PropTypes from "prop-types";
+import EditActionBar from "components/Todo/EditCard/edit-action-bar.component";
+import { ENTER } from "common/constants";
+
+const EditCard = ({
+  todo,
+  handleEditTask,
+  handleCompleteTask,
+  handleEditCancelTask,
+}) => {
+  const { id, task } = todo;
+  const textRef = useRef(null);
+  const [editedTask, setEditData] = useState(task);
+
+  const handleChange = (e) => {
+    setEditData(e.target.value);
+  };
+  const handleKeyDown = (e) => {
+    if (e.key === ENTER) {
+      handleEditTask(id, editedTask);
+    }
+  };
+
+  const setFocusAndSelection = () => {
+    textRef.current.focus();
+    textRef.current.selectionStart = textRef.current.value.length;
+    textRef.current.selectionEnd = textRef.current.value.length;
+  };
+
+  useEffect(() => {
+    setFocusAndSelection();
+  }, [task]);
+
+  return (
+    <>
+      <textarea
+        name="todo_input"
+        id="todo_input"
```

Comment:
follow naming convention of HTML attributes.

```suggestion
        id="todo-input"
```